src/cartridges/int_adyen_SFRA/cartridge/adyen/analytics/analyticsHook.js

```diff
@@ -0,0 +1,107 @@
+const URLRedirectMgr = require('dw/web/URLRedirectMgr');
+const analyticsEvent = require('*/cartridge/adyen/analytics/analyticsEvents');
+const analyticsConstants = require('*/cartridge/adyen/analytics/constants');
+const constants = require('*/cartridge/adyen/config/constants');
+const {
+  isAdyenAnalyticsEnabled,
+} = require('*/cartridge/adyen/utils/adyenConfigs');
+
+function getPath(req) {
+  return req.path.split('/').pop().split('-');
+}
+
+function redirectUrlStartRule() {
+  const origin = URLRedirectMgr.redirectOrigin;
+  return origin.match(constants.APPLE_DOMAIN_URL);
+}
+
+function isUsedByAdyen(req) {
+  const path = getPath(req);
+  const allControllers = [
+    { name: 'account', routes: [{ name: 'Show' }] },
+    { name: 'adyen' },
+    { name: 'cart', routes: [{ name: 'MiniCartShow' }] },
+    { name: 'checkout', routes: [{ name: 'Begin' }] },
+    { name: 'checkoutServices', routes: [{ name: 'PlaceOrder' }] },
+    { name: 'order', routes: [{ name: 'Confirm' }] },
+    {
+      name: 'paymentInstruments',
+      routes: [
+        { name: 'AddPayment' },
+        { name: 'SavePayment' },
+        { name: 'DeletePayment' },
+      ],
+    },
+    {
+      name: 'redirectUrl',
+      routes: [{ name: 'Start', rule: redirectUrlStartRule }],
+    },
+  ];
+  return allControllers.some((controller) => {
+    if (path[0].toLowerCase() === controller.name.toLowerCase()) {
+      if (!controller.routes) {
+        return true;
+      }
+      const route = controller.routes.find(
+        (eachRoute) => path[1].toLowerCase() === eachRoute.name.toLowerCase(),
+      );
+      if (typeof route?.rule === 'function') {
+        return !!route.rule();
+      }
+      return !!route;
+    }
+    return false;
+  });
+}
+
+function onStartHandler(req) {
+  const path = getPath(req);
+  if (isUsedByAdyen(req) && isAdyenAnalyticsEnabled()) {
+    analyticsEvent.createAnalyticsEvent(
+      session.sessionID,
+      path.join('-'),
+      analyticsConstants.eventType.START,
+      analyticsConstants.eventStatus.EXPECTED,
+      analyticsConstants.eventCode.INFO,
+    );
+  }
+}
+
+function onCompleteHandler(req, res) {
+  const path = getPath(req);
+  const hasError = !!res.viewData?.error;
```

Comment:
This const will not be always true on the negative scenarios of our Adyen related calls (e.g. if /paymentMethods call fails, the error will be found under res.viewData.AdyenPaymentMethods.error, while if Adyen-ShippingMethods fails the error will be found under res.viewData.errorMessage), hence, instead of receiving UNEXPECTED event we receive EXPECTED event. 
Ideally we should come up with a way of handling the error in same way in all catch blocks.

Reply:
Created [SFI-1007](https://youtrack.is.adyen.com/issue/SFI-1007/Propagate-adyenError-in-controllers-on-res) for the error propagation and handling.